Source/WebKit/GPUProcess/graphics/RemoteDisplayListRecorder.cpp

```diff
@@ -514,9 +514,9 @@ void RemoteDisplayListRecorder::setSharedVideoFrameSemaphore(IPC::Semaphore&& se
     m_sharedVideoFrameReader.setSemaphore(WTFMove(semaphore));
 }
 
-void RemoteDisplayListRecorder::setSharedVideoFrameMemory(const SharedMemory::Handle& handle)
+void RemoteDisplayListRecorder::setSharedVideoFrameMemory(SharedMemory::Handle&& handle)
```

Comment:
Isn't SharedMemory::Handle essentially just an integer (or 2 with the size)? If so, it doesn't seem like there would be much benefit to moving it around?

Reply:
This patch started when I asked @kkinnunen-apple about what the behavior of `SharedMemory::map` was supposed to be with regards to the `handle` passed in. The behavior was not consistent between the different backend implementations, Win, Unix, Cocoa. I found this out when enabling GPU Process on PlayStation where the Unix variant of `SharedMemory::createHandle` asserted but the Windows one was fine.

So @kkinnunen-apple expressed a plan to refactor `SharedMemory::map` to take a `Handle&&` but wasn't available to work on it till later. Since I was broken I took a stab at this. I THINK I replicated their plan but I'll let them chime in.

Reply:
👍🏻 This is more about transferring ownership when optimizing copies. Seems fine.